api/internal/crawl/crawler/crawler.go

```diff
@@ -29,7 +29,7 @@ type Crawler interface {
 	// Crawl returns when it is done processing. This method does not take
 	// ownership of the channel. The channel is write only, and it
 	// designates where the crawler should forward the documents.
-	Crawl(ctx context.Context, output chan<- CrawledDocument) error
+	Crawl(ctx context.Context, output chan<- CrawledDocument, seen map[string]struct{}) error
```

Comment:
can we make `seen` a non-primitive type?
that way we can distinguish it from branchMap more easily.  plus type safety, etc.

```
type seenMap map[string]bool

func (of seenMap) alreadySaw(arg string) bool {
	_, ok := of[arg]
	return ok
}

func (of seenMap) recognize(arg string) {
	of[arg] = true
}

func newSeenMap() seenMap {
	return make(map[string]bool)
}
```

Reply:
Done